### PR TITLE
CT-3753 Add columns in warehouse for vetting process

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to add 3 columns for tracking vetting performance
## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3753

### CODE REVIEW

File: app/models/warehouse/case_report.rb [hunk 1]
```diff
@@ -186,6 +186,10 @@ def process_offender_sar(kase, case_report)
         case_report.third_party_company_name = kase.third_party_company_name
         case_report.number_of_exempt_pages = kase.number_exempt_pages
         case_report.number_of_final_pages = kase.number_final_pages
+        case_report.number_of_days_for_vetting = kase.number_of_days_for_vetting
+        user_for_vetting = kase.user_dealing_with_vetting
+        case_report.user_dealing_with_vetting = user_for_vetting.nil? ? nil : user_for_vetting.full_name
```
Comment (willmcb) [inline app/models/warehouse/case_report.rb:191]: I think you can call `user_for_vetting&.full_name` and `user_for_vetting&.id` here (safe navigation operator, see: https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/)

As the method returns `nil` and `nil&.full_name` ==> `nil`

**Review by willmcb — APPROVED** [review]
LGTM - few minor points you may want to consider :)

File: app/models/case/sar/offender.rb [hunk 2]
```diff
@@ -292,6 +296,38 @@ def first_prison_number
     prison_number.gsub(/[,]/, ' ').split(' ').first.upcase
   end
 
+  def number_of_days_for_vetting
+    # Get the start date and end date for the vetting process
+    start_date_for_vetting = nil
+    end_date_for_vetting = nil
+    self.transitions.each do | transition|
+      if transition.event == VETTING_IN_PROCESS_EVENT
+        start_date_for_vetting = transition.created_at.to_date
```
[the remark this answers is not in the document]
Comment (ymao2) [reply app/models/case/sar/offender.rb:305]: created_date is datetime type,

File: app/models/case/sar/offender.rb [hunk 3]
```diff
@@ -292,6 +296,38 @@ def first_prison_number
...
+      if transition.event == READY_FOR_COPY_EVENT
+        end_date_for_vetting = transition.created_at.to_date
+      end
+    end
+    end_date_for_vetting = end_date_for_vetting || Date.today
+    # Calculate the days taken for vetting process
+    days = nil
+    if start_date_for_vetting
+      days = start_date_for_vetting.business_days_until(end_date_for_vetting, true) 
+      days > 0 ? days : nil
```
[the remark this answers is not in the document]
Comment (ymao2) [reply app/models/case/sar/offender.rb]: yes, you are right, I  changed into 0

File: app/models/warehouse/case_report.rb (hunk 1 above)
Comment (ymao2) [reply app/models/warehouse/case_report.rb]: didn't know this,  nice, I changed to the above one

**Pull request merged**